src/altinn-app-frontend/src/components/advanced/AddressComponent.tsx

```diff
@@ -289,35 +288,35 @@ export function AddressComponent({
       className='address-component'
       key={`address_component_${id}`}
     >
-      <AddressLabel
-        labelKey={'address_component.address'}
-        id={`address_address_${id}`}
-        language={language}
-        required={required}
-        readOnly={readOnly}
-        labelSettings={labelSettings}
-      />
-      <input
-        id={`address_address_${id}`}
-        className={cn('form-control', {
-          'validation-error': allValidations.address.errors.length,
-          disabled: readOnly,
-        })}
-        value={address}
-        onChange={updateField.bind(null, AddressKeys.address, false)}
-        onBlur={updateField.bind(null, AddressKeys.address, true)}
-        onPaste={() => onAddressPaste()}
-        readOnly={readOnly}
-        required={required}
-      />
-      {allValidations?.[AddressKeys.address]
-        ? renderValidationMessagesForComponent(
-            allValidations[AddressKeys.address],
-            `${id}_${AddressKeys.address}`,
-          )
-        : null}
+      <div>
+        <AddressLabel
+          labelKey={'address_component.address'}
+          id={`address_address_${id}`}
+          language={language}
+          required={required}
+          readOnly={readOnly}
+          labelSettings={labelSettings}
+        />
+        <TextField
+          id={`address_address_${id}`}
+          isValid={allValidations.address.errors.length === 0}
+          value={address}
+          onChange={updateField.bind(null, AddressKeys.address, false)}
+          onBlur={updateField.bind(null, AddressKeys.address, true)}
+          onPaste={() => onAddressPaste()}
+          readOnly={readOnly}
+          required={required}
+        />
+        {allValidations?.[AddressKeys.address]
+          ? renderValidationMessagesForComponent(
+              allValidations[AddressKeys.address],
+              `${id}_${AddressKeys.address}`,
+            )
+          : null}
+      </div>
+
       {!simplified && (
-        <>
+        <div>
```

Comment:
Any reason for converting this fragment to a div? I prefer using fragments instead of divs (to avoid cluttering the DOM) unless we're actually styling the div somehow. And if we're styling it, we should be adding a class to it. Maybe it's just a question of personal taste..

Reply:
Same thing on line 403

Reply:
Yes, the reason is to group the items properly so we can take advantage of the `gap` css property that is set on the common parent. This makes it easier to lay out the items with correct spacing between them. The old implementation used `margin-bottom` instead, which caused extra space before the next content. This could f.ex be solved by setting no margin on the last item instead, but using `gap` is a lot cleaner. The only "downside" is that the items need to be grouped properly.

Reply:
Ah - so you'd see a visual difference if you'd try removing them, then. Fine by me! :+1: